### PR TITLE
MRG: Add "descriptions=False" kwarg to get_builtin_montages, and use get_builtin_montages in sensor location tutorial

### PR DESCRIPTION
**Pull request by hoechenberger**
The motivation was to make it easier for users to programmtically find out which standard montages are available.

WIP:

- [x] tests
- [x] changelog

### CODE REVIEW

**Comment by hoechenberger** [comment]
Rendered tutorial:
https://output.circle-artifacts.com/output/job/1b5bff54-068d-44c2-a81f-4787fb09c585/artifacts/0/dev/auto_tutorials/intro/40_sensor_locations.html#working-with-built-in-montages

**Comment by hoechenberger** [comment]
cc @drammock

**Comment by drammock** [comment]
Tutorial section looks great but remove the emoji bullets please

File: tutorials/intro/40_sensor_locations.py [hunk 1]
```diff
@@ -58,17 +58,12 @@
...
-montages = sorted(path.name for path in montage_dir.iterdir())
-print(
-    '\n'
-    'BUILT-IN MONTAGES\n'
-    '================='
-)
-print('\n'.join(montages))
+builtin_montages = mne.channels.get_builtin_montages(descriptions=True)
+for montage_name, montage_description in builtin_montages:
+    print(f'🧠 {montage_name}: {montage_description}')
```
Comment (agramfort) [inline tutorials/intro/40_sensor_locations.py:66]:
```suggestion
    print(f'{montage_name}: {montage_description}')
```

**Review by agramfort — COMMENTED** [review]
+1 for MRG after removing the emoji

thx @hoechenberger

File: doc/changes/latest.inc [hunk 2]
```diff
@@ -93,6 +93,8 @@ Enhancements
 
 - All functions and methods that plot topographic maps for EEG (2D projections of the EEG sensor locations) now accept the parameter value ``sphere='eeglab'`` to lay out the sensors with respect to the head circle in a similar way to how EEGLAB does, i.e., T7/T8 and Fpz/Oz are placed directly on the circle for template montages (:gh:`10572` by `Richard Höchenberger`_)
 
+- :func:`mne.channels.get_builtin_montages` gaines a new parameter, ``descriptions``, which allows to retrieve the descriptions of the montages in addition to their names (:gh:`10373` by `Richard Höchenberger`_)
```
Comment (hoechenberger) [inline doc/changes/latest.inc:96]:
```suggestion
- :func:`mne.channels.get_builtin_montages` gained a new parameter, ``descriptions``, which allows to retrieve the descriptions of the montages in addition to their names (:gh:`10373` by `Richard Höchenberger`_)
```

File: mne/channels/montage.py [hunk 3]
```diff
@@ -65,16 +176,35 @@ def _check_get_coord_frame(dig):
...
+
+    The names of the montages can be passed to :func:`make_standard_montage`.
+
+    Parameters
+    ----------
+    descriptions : bool
+        Whether to return not only the montage names, but also their
+        corresponding descriptions. If ``True``, a list of tuples is returned,
+        where the first tuple element is the montage name and the second is
+        the montage description.
```
Comment (hoechenberger) [inline mne/channels/montage.py:190]:
```suggestion
        the montage description. If ``False`` (default), only the names are
        returned.
```

**Review by agramfort — APPROVED** [review]

**Comment by hoechenberger** [comment]
@agramfort I did one more change to the `set_montage()` API docs. Let's wait for the Circle rendering to come back so we can check if it all looks okay!

**Comment by hoechenberger** [comment]
Rendered API docs, just spotted a typo but otherwise it's good I believe!

https://output.circle-artifacts.com/output/job/a7911bac-2d21-4c3c-9b39-698913816713/artifacts/0/dev/generated/mne.BaseEpochs.html?highlight=set_montage#mne.BaseEpochs.set_montage

File: mne/utils/docs.py [hunk 4]
```diff
@@ -1854,11 +1854,16 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
...
 montage : None | str | DigMontage
-    A montage containing channel positions. If str or DigMontage is
-    specified, the channel info will be updated with the channel
-    positions. Default is None. For valid :class:`str` values see documentation
-    of :func:`mne.channels.make_standard_montage`. See also the documentation
-    of :class:`mne.channels.DigMontage` for more information.
+    A montage containing channel positions. If a string or
+    :class:`~mne.channels.DigMontage` is
+    specified, the existing channel information will be updated with the
+    channel positions in from montage. Valid strings are the names of the
```
Comment (hoechenberger) [inline mne/utils/docs.py:1860]:
```suggestion
    channel positions from the montage. Valid strings are the names of the
```

**Pull request merged**